### DISCUSSION TITLE
[Bug]: 12 Step Meeting PDF Generator Broken as a result of recent changes in TSML

### DISCUSSION THREAD

**Discussion by marcinyorkpa** [code4recovery/12-step-meeting-list#548] [General]
### Contact Details

webmaster@york-pa-aa.org

### Website With Issue

https://york-pa-aa.org

### What happened?

We use the "12 Step Meeting PDF Generator" WordPress plugin, from https://github.com/cdtoews/12-step-meeting-pdf, that works in concert with the 12 Step Meeting List.

Starting sometime recently, that PDF Generator plugin began throwing this error:

```
 Fatal error: Uncaught Error: Call to undefined function tsml_define_strings()
in /hermes/bosnaweb23a/b2708/ipg.yaiweb/york-pa-aa/york-pa-aa/wp-content/plugins/12-step-meeting-pdf-generator/includes/admin-gen.php on line 209

Call stack:

    tsmp_gen_page()
    wp-includes/class-wp-hook.php:303
    WP_Hook::apply_filters()
    wp-includes/class-wp-hook.php:327
    WP_Hook::do_action()
    wp-includes/plugin.php:470
    do_action()
    wp-admin/admin.php:259
    require_once()
    wp-admin/edit.php:10
```

I realize that another plugin breaking isn't your problem... but it looks like that plugin isn't actively being worked on, and it looks like it is tightly coupled to the `tsml_define_strings()` function in the TSML.

Do you have any advice on how to get help with this? I'm not sure what the relationship is between the PDF Generator Plugin and TSML, but perhaps if you know each other and talk, y'all could see if you could fix this? If not, I'll see if I can figure out how to get support for that plugin.

Thanks!!!

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

**Comment by petermixt** [comment]
the "12 Step Meeting PDF Generator" WordPress plugin -- the current menu returns:  Error thrown
Call to undefined function tsml_define_strings()

Is this caused by updates to your 12 Step Meeting List?

**Reply by tech2serve** [reply]
I don't know yet. The first issue was due to a domain settings goof on our part.

Please look in the dev console and post the complete error here.

Thanks.

**Comment by tech2serve** [comment]
@marcinyorkpa I'm transferring this to a Discussion.

**Comment by tech2serve** [comment]
@marcinyorkpa and @petermixt --

It is possible that the 12 Step Meeting List PDF plugin by Chris T (@cdteows) may not be compatible with our new PDF Generator Service. 

Our service will remain actively developed by Code for Recovery, especially with community help and feedback. 

As Chris' GitHub page doesn't have a mechanism for discussions or feedback, there isn't a clear way to notify him of the issue. Perhaps if someone knows him and can reach out?

If either of you have coding experience, you could try working off a fork of his repo.

The Code for Recovery developers will also discuss this more to see if there are other options.

In the meantime, you might try disabling the plugin and seeing if our new PDF service meets your needs.

**Reply by cdtoews** [reply]
I just enabled discussion on that github repo.  https://github.com/cdtoews/12-step-meeting-pdf/discussions
Comments can be put there,
or support at the wordpress plugin page. https://wordpress.org/support/plugin/12-step-meeting-pdf-generator/
The "12 step meeting list pdf" plugin is something I do in my spare time, so I can only fix things as my personal time permits.

**Comment by petermixt** [comment]
Not being able to contact a developer is always an issue when updates are needed.
I''m looking forward to using your new PDF generator when it works.  
My JSON file is correct and open independently on my browser... but returns an error as I have posted as a bug.
Here is a screen shot of what I get.
![Screenshot 2021-10-03 at 17-27-43 PDF Generator](https://user-images.githubusercontent.com/38595503/135780142-7cd62136-1b85-4a25-9409-5389b6473fa5.png)

Thank you,
Pete S.
PeterMixt

**Reply by joshreisner** [reply]
hi peter, i created an issue for this in the pdf repo. we’ll take a look soon!

**Reply by cdtoews** [reply]
Is that image another PDF generator for data inside TSML? 
Is that a reasonable replacement for this one: https://wordpress.org/plugins/12-step-meeting-pdf-generator/ ?

**Reply by joshreisner** [reply]
hey chris, it's an external PDF service that i developed - but it doesn't really do the same stuff as your plugin. no tri-fold, no custom HTML, pretty different format.

**Comment by joshreisner** [comment]
Hi @petermixt as for the original error, I've confirmed that yes we removed the `tsml_define_strings` function from our plugin as a part of code cleanup. It's unfortunate that 12 Step Meeting PDF Generator is calling this function without checking that it exists first. 

In my testing it looks like this function call isn't necessary and the PDF works ok without it. So if you add the following code to your theme or child theme you should be good to go:

```
if (!function_exists('tsml_define_strings')) {
  function tsml_define_strings()
  {
  }
}
```

**Reply by joshreisner** [reply]
I created a PR to remove this function: https://github.com/cdtoews/12-step-meeting-pdf/pull/1

**Comment by petermixt** [comment]
Thank you gentlemen. You are rock stars.
Pete S

On Sun, Oct 3, 2021 at 10:22 PM Josh Reisner ***@***.***>
wrote:

> I created a PR to remove this function: cdtoews/12-step-meeting-pdf#1
> <https://github.com/cdtoews/12-step-meeting-pdf/pull/1>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/548#discussioncomment-1422792>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJGOXL52H3RTVI5JYWCKCXTUFE2YBANCNFSM5FIGUY3Q>
> .
>
-- 
Sent from Pete’s iPhone
Trust the Process...
